cmd/contour/shutdownmanager.go

```diff
@@ -98,17 +101,44 @@ func (s *shutdownmanagerContext) healthzHandler(w http.ResponseWriter, r *http.R
 // file to understand if it is safe to terminate. The file-based approach is used since the process in which
 // the kubelet calls the shutdown command is different than the HTTP request from Envoy to /shutdown
 func (s *shutdownmanagerContext) shutdownReadyHandler(w http.ResponseWriter, r *http.Request) {
+	// For testing, the filepath and check period can be overridden in the context
+	// Otherwise, use the module constants
+	var signalFilepath string
+	if s.shutdownReadyFile != "" {
+		signalFilepath = s.shutdownReadyFile
+	} else {
+		signalFilepath = shutdownReadyFile
+	}
+	var signalFileCheckDuration time.Duration
+	if s.shutdownReadyCheckInterval != 0 {
+		signalFileCheckDuration = s.shutdownReadyCheckInterval
+	} else {
+		signalFileCheckDuration = shutdownReadyCheckInterval
+	}
+
+	l := s.WithField("context", "shutdownReadyHandler")
+	ctx := r.Context()
 	for {
-		_, err := os.Stat(shutdownReadyFile)
-		if err != nil {
+		_, err := os.Stat(signalFilepath)
+		if err == nil {
```

Comment:
Not a huge deal, but I might reorder this to handle the error cases first, which is more common in go (I'd guess that not writing it that way in the first place contributed to the regression here) - so something like:

```go
if err != nil {
    if os.IsNotExist(err) {
        ...
    } else {
       ...
    }

    // wait & continue
}

// no error: ready for shutdown, write response
```

Reply:
Not exactly what you suggested, but I reordered the if cases.